src/server/game/Entities/Player/Player.cpp

```diff
@@ -27594,4 +27594,124 @@ bool Player::IsPetDismissed()
     return false;
 }
 
+/* Premium character interaction.
+ * These functions will allow modules to access the new generic
+ * premium tables designed to help modules that interact with
+ * players and characters without the need to create other tables.
+ */
+int Player::GetCharacterPremiumLevel(uint32 guid)
+{
+    PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_PREMIUM_CHARACTER_PREMIUM_LEVEL);
```

Comment:
You should create a new container 
like:
`GuidSet m_premiumCharacters;`
`GuidSet m_premiumAccounts;`

and fill them up on Server start (and reloadable via command).

**If there is a specific level you can also create an UnorderedMap which hold AccountId + Level.**

This will save unneeded database calls and you only need to have the function to be something like: (Quickly coded for showing you, might not be copy pastable)
```cpp
        bool IsPremiumCharacter(ObjectGuid const& guid) const
        {
            return m_premiumCharacters.find(guid) != m_premiumCharacters.end();
        }
```